pkg/kudoctl/cmd/output/output.go

```diff
@@ -4,18 +4,38 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"strings"
 
 	"sigs.k8s.io/yaml"
 )
 
+// OutputType specifies the type of output a command produces
+type Type string
+
 const (
-	TypeYAML = "yaml"
-	TypeJSON = "json"
+	// StringValueType is used for parameter values that are provided as a string.
+	TypeYAML Type = "yaml"
+
+	// ArrayValueType is used for parameter values that described an array of values.
+	TypeJSON Type = "json"
+
+	InvalidOutputError = "invalid output format, only support 'yaml' or 'json' or empty"
 )
 
-func WriteObjects(objs []interface{}, outputType string, out io.Writer) error {
-	if strings.ToLower(outputType) == TypeYAML {
+func ValidateType(outputType Type) error {
+	switch outputType {
+	case TypeYAML:
```

Comment:
if we define all supported outputs somewhere in here we might use it in this method as well

Reply:
Done 👍